File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n = nums.size();
        map<int,int> m;
        for(int i=0; i<n; i++) {
            int ans = target - nums[i];
            if(m.find(ans) != m.end()) {
                return {m[ans],i};
            }
            m[nums[i]] = i;
        }
        return {-1,-1};
    }
};

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> m = new HashMap<>();
        for(int i=0; i<nums.length; i++) {
            int x = target - nums[i];
            if(m.containsKey(x) && m.get(x) != i) return new int[]{i,m.get(x)};
            m.put(nums[i],i);
        }
        return new int[] {-1,-1};
    }
}